plugins/inputs/icinga2/icinga2.go

```diff
@@ -123,39 +195,74 @@ func (i *Icinga2) Gather(acc telegraf.Accumulator) error {
 		i.client = client
 	}
 
-	requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
-
-	// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
-	//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
-	if i.ObjectType == "services" {
-		requestURL += "&attrs=host_name"
+	// For backward config compatibility
+	if i.ObjectType != "" {
+		i.Objects = []string{i.ObjectType}
 	}
 
-	address := fmt.Sprintf(requestURL, i.Server, i.ObjectType)
+	// Collect /v1/objects
+	for _, objectType := range i.Objects {
+		requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
 
-	req, err := http.NewRequest("GET", address, nil)
-	if err != nil {
-		return err
-	}
+		// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
+		//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
+		if objectType == "services" {
+			requestURL += "&attrs=host_name"
+		}
 
-	if i.Username != "" {
-		req.SetBasicAuth(i.Username, i.Password)
-	}
+		address := fmt.Sprintf(requestURL, i.Server, objectType)
 
-	resp, err := i.client.Do(req)
-	if err != nil {
-		return err
-	}
+		resp, err := i.IcingaRequest(address)
+		if err != nil {
+			return err
+		}
 
-	defer resp.Body.Close()
+		result := ResultObject{}
+		err = json.NewDecoder(resp.Body).Decode(&result)
+		if err != nil {
+			return err
+		}
+		err = resp.Body.Close()
+		if err != nil {
+			return err
+		}
```

Comment:
For the other result processings, You created a specific method (eg `ParseCIBResponse`). Why not here?

Reply:
I've tried not to touch the original code, but that didn't really work out.